src/misc/ErrorHandlerImpl.ts

```diff
@@ -174,62 +175,65 @@ function logoutIfNoPasswordPrompt() {
 }
 
 export async function reloginForExpiredSession() {
-	if (!loginDialogActive) {
-		const { logins, loginFacade, secondFactorHandler, credentialsProvider, sqlCipherFacade } = locator
-		// Make sure that partial login part is complete before we will try to make a new session.
-		// Otherwise we run into a race condition where login failure arrives before we initialize userController.
-		await logins.waitForPartialLogin()
-		console.log("RELOGIN", logins.isUserLoggedIn())
-		const oldSessionType = logins.getUserController().sessionType
-		const userId = logins.getUserController().user._id
-		const mailAddress = neverNull(logins.getUserController().userGroupInfo.mailAddress)
-		// Fetch old credentials to preserve database key if it's there
-		const oldCredentials = await credentialsProvider.getCredentialsByUserId(userId)
-		const sessionReset = loginFacade.resetSession()
-		loginDialogActive = true
+	if (loginDialogActive) {
+		return
+	}
+	const { logins, loginFacade, secondFactorHandler, credentialsProvider, sqlCipherFacade, cacheStorage } = locator
+	// Make sure that partial login part is complete before we will try to make a new session.
+	// Otherwise we run into a race condition where login failure arrives before we initialize userController.
+	await logins.waitForPartialLogin()
+	console.log("RELOGIN", logins.isUserLoggedIn())
+	const oldSessionType = logins.getUserController().sessionType
+	const userId = logins.getUserController().user._id
+	const mailAddress = assertNotNull(logins.getUserController().userGroupInfo.mailAddress, "could not get mailAddress from userGroupInfo")
```

Comment:
This show function is getting so complicated already that I wonder if we should shove it into `LoginFacade` somehow, maybe in two parts but probably later

Reply:
I don't think this part will be redone together with the login screen tbh

Reply:
I thought we're talking about a redo of the login flow not only UI